### PR TITLE
Addingauthplaces and fixes over the weekend

### PR DESCRIPTION
**Pull request by marcel-epp**
- bugfix: scroll to top now on every page
- bugfix: the comments will also be deleted in the users array. No Zombie comments anymore.
- bugfix: the user can edit and delete his own comments, identified by the user id and not the name. This solved the "Thomas Müller problem" from two different persons with the same name
- feature: you can create places now
- feature: the own created places are listed in the profile
- feature: the own place can be deleted, all associated comments will also be deleted, and from the userComments/favoritePlaces array the comment is also be deleted. The image on cloudinary is also be deleted.
- all places have images now on cloudinary
- all places have now dummy text for the description and review
- optimized the website for a presentation on a 13 Inch Macbook

### CODE REVIEW

File: components/Partials/Layout.js [hunk 1]
```diff
@@ -17,6 +18,18 @@ export default function Layout({ children, togglePageActive }) {
       </Head>
       <Header togglePageActive={togglePageActive} />
       <StyledMain>{children}</StyledMain>
+      <StyledFooterImage
+        src="/images/footer-70.png"
+        sizes="100vw"
+        style={{
+          width: "100%",
+          height: "auto",
+        }}
```
Comment (MerleMak) [inline components/Partials/Layout.js:27]: no inline styling, especially in components, that are already styled compoents :D

File: pages/api/user/[id].js [hunk 2]
```diff
@@ -7,7 +7,7 @@ export default async function handler(request, response) {
   const { id } = request.query;
 
   const user = await User.findById(id).populate({
-    path: "favoritePlaces",
+    path: "favoritePlaces createdPlaces",
```
Comment (MerleMak) [inline pages/api/user/[id].js:10]: the path shouldn't contain white spaces. 
Do you mean:
```suggestion
    path: "createdPlaces",
```

File: components/Forms/FormImageUpload.js [hunk 3]
```diff
@@ -28,19 +37,113 @@ const UploadImage = ({ isImage, setIsImage }) => {
...
-    </div>
+    <>
+      <StyledUpload>
+        {image && (
+          <StyledUploadImage
+            sizes="100vw"
+            style={{
+              width: "100%",
+              height: "auto",
+            }}
```
Comment (MerleMak) [inline components/Forms/FormImageUpload.js:58]: no inline styling :)

**Review by MerleMak — COMMENTED** [review]
Hi Marcel, 
I left you some comments, but one thing that is really really important:
Please separate the different concerns (different US) into different PRs.
That drastically improves the quality of the code reviews. It's really important that you do that, because for many companies its a requirement for their quality assurance in the code.

But otherwise the code looks good 👍🏼

File: pages/api/user/[id].js (hunk 2 above)
Comment (marcel-epp) [reply pages/api/user/[id].js:10]: Okay, but i need this to populate favoritePlaces and createdPlaces.

Comment (marcel-epp) [reply pages/api/user/[id].js:10]: I changed the code to this, i don't have a better solution at the moment...

`const user = await User.findById(id)
    .populate({
      path: "favoritePlaces",
      populate: { path: "reviews activities" },
    })
    .populate({
      path: "createdPlaces",
      populate: { path: "reviews activities" },
    });`

**Pull request merged**